### DISCUSSION TITLE
리더보드 GetRangeRank가 실행이 안되는 현상입니다,,

### DISCUSSION THREAD

**Discussion by lhj5709** [naverz/zepeto-studio-kor#471] [Issue Report (Legacy)]
리더보드 예제를 따라하면서 적용해보고 있는데요,,
GetRangeRank() 함수가 작동이 안되는 현상이 있습니다..

코드입니다.

`Start() {    

        LeaderboardAPI.GetAllLeaderboards(this.OnResultGetAllLeaderBoard, this.OnError);

        this.StartCoroutine(this.TestCoroutine());
    }

    *TestCoroutine(){
        yield null;

        yield new WaitForSeconds(5);
        this.SetLeaderboardData("ClearTime", 77.23);

        yield new WaitForSeconds(5);
        this.GetLeaderboardData("ClearTime");
    }
    

    // Get Leaderboard
    public GetLeaderboardData(leaderboardName: string){

        let leaderboard_id = "";
        if(leaderboardName == "ClearTime"){
            leaderboard_id = this.leaderboard_ID;
        }
        else{
            console.error("Can not find leaderboard id");
            return;
        }

        LeaderboardAPI.GetRangeRank(leaderboard_id, 1, 10, this.resetRule, false, this.OnResultGetLeaderboard, this.OnError);
        
        Debug.Log("Get2222");
        
    }

    private OnResultGetLeaderboard(result: GetRangeRankResponse){

        console.log(`result.isSuccess: ${result.isSuccess}`);
       if (result.rankInfo.myRank) {
           console.log(`member: ${result.rankInfo.myRank.member}, rank:  ${result.rankInfo.myRank.rank}, score:${result.rankInfo.myRank.score}, name: ${result.rankInfo.myRank.name}`);
       }
 
       if (result.rankInfo.rankList) {
           for (let i = 0; i < result.rankInfo.rankList.length; ++i) {
               var rank = result.rankInfo.rankList.get_Item(i);
               console.log(`i: ${i}, member: ${rank.member}, rank: ${rank.rank}, score: ${rank.score}, name: ${result.rankInfo.myRank.name}`);
           }
       }
        
    }

    // Get leaderboard End

    // Set Leaderboard
    public SetLeaderboardData(leaderboardName : string, record : float){

        if(leaderboardName == "ClearTime"){
            LeaderboardAPI.SetScore(this.leaderboard_ID, record, (result: SetScoreResponse)=>{

                console.log(`[SET] result.isSuccess: ${result.isSuccess}`);

            }, this.OnError);
        }

    }`

![image](https://user-images.githubusercontent.com/20422451/157182552-457560d5-a6a8-47cb-9864-ade2a3fb1b6e.png)

로그를 보면 Result를 받는 부분이 실행이 안됩니다,, 해결방법이 있을까요?

기존 디스커션을 검색해본 결과,  람다식으로 하라는 답변을 보고 람다식으로 해봐도 똑같이 실행이 안됐습니다,,

**Comment by minjujuu** [comment]
아래 코드로 해서 로그 출력 확인되는데, 한 번 시도해보시면 좋을 것 같아 공유 드립니다!

```
            LeaderboardAPI.GetRangeRank(this.leaderboardId, this.startRank, this.endRank, this.resetRule, false, ((result: GetRangeRankResponse) => {

            console.log(`result.isSuccess: ${result.isSuccess}`);
            if (result.rankInfo.myRank) {
                console.log(`my rank : ${result.rankInfo.myRank.rank.toString()}`);
                console.log(`member: ${result.rankInfo.myRank.member}, rank: ${result.rankInfo.myRank.rank}, score: ${result.rankInfo.myRank.score}, name: ${result.rankInfo.myRank.name}`);
                var resultInfo = result.rankInfo;
                this.rank = resultInfo.myRank.rank.toString();
            }

            if (result.rankInfo.rankList) {
                for (let i = 0; i < result.rankInfo.rankList.length; ++i) {
                    var rank = result.rankInfo.rankList.get_Item(i);
                    console.log(`i: ${i}, member: ${rank.member}, rank: ${rank.rank}, score: ${rank.score}`);
                }
            }
        }), this.OnError);
```

![image](https://user-images.githubusercontent.com/79794414/157188590-d5e6d068-48e4-491f-a267-2b0c2579ff39.png)

**Reply by lhj5709** [reply]
![image](https://user-images.githubusercontent.com/20422451/157189433-334176a5-09fa-4d3a-8817-5256bc5b834a.png)

네,, 위 코드처럼 람다식으로 변경해서 해봐도 똑같이 출력이 안되면 뭘 확인해봐야할까요? ㅠㅠ 너무 답답합니다

**Reply by minjujuu** [reply]
음ㅠ leaderboard id는 Project settings - Zepeto - Zepeto Leaderboard 등록된 걸로 잘 들어가고 있을까요?

**Reply by lhj5709** [reply]
![image](https://user-images.githubusercontent.com/20422451/157192748-3f16afc8-c783-4012-9b7f-22e4919676f8.png)
![image](https://user-images.githubusercontent.com/20422451/157192602-45cc816a-ede6-4d18-a75d-3cd02ce4a6e5.png)

넵,,, 카피해서 그대로 붙여 넣었습니다 ㅠㅠ

**Reply by lhj5709** [reply]
![image](https://user-images.githubusercontent.com/20422451/157193004-2848bca9-63e9-4353-93db-ab1e420c1405.png)

start에서 리더보드 체크를 했을 때도 조회가 잘 되는데 딱 RankingData만 가져오지를 못하네요 ㅠㅠ

**Reply by minjujuu** [reply]
그럼 혹시 World Id가 스튜디오 콘솔에 등록된 월드의 Id로 등록되었을까요?
![image](https://user-images.githubusercontent.com/79794414/157788549-bc80ef36-b8ac-41fd-870a-cb4df99f7c95.png)
![image](https://user-images.githubusercontent.com/79794414/157788634-a2cc101c-668d-46fc-aae6-a723354cd72c.png)

**Reply by Sonak0930** [reply]
혹시 이 문제 해결하셨을까요?
